### ISSUE TITLE
Advanced search won't search on extra long project title names

### ISSUE THREAD

**Issue by carolynrosner** [ua-snap/nccwsc-projects#84]
In the advanced project search, you can search for full title names successfully if the title is pretty short (e.g. Big Sagebrush Response to Wildfire and Invasive Grasses in the 21st Century), but if the title is longer (e.g. How Will Coldwater Fish Survive in a Warming Future? Identifying Life-Stage Specific Use of Coldwater Refugia in the Klamath Basin and Willamette River), the search doesn't work

**Comment by BobTorgerson** [comment]
For this issue and #85 we are at the mercy of what we have returned to us from the ScienceBase cache API and what we can provide to the API to produce valid search results. I've discovered that there is really no limit to how large a project title could be as long as that title doesn't contain any special characters such as a **?** as in the long title above. These special characters are being properly encoded to their URI safe entity before being passed to the ScienceBase API on our end, but the search functionality on that end doesn't accept those URI entities as part of the query so it is not returning anything. However, if you try searching for: **How Will Coldwater Fish Survive in a Warming Future Identifying Specific Use of Coldwater Refugia in the Klamath Basin and Willamette River** that project pops right up.

For both this and #85, we will need to have Dell accept those special characters on their end for the search to work properly. For example, here is a working search query with %20 representing spaces in the below URL:

https://my.usgs.gov/casc-service/search?query=How%20Will%20Coldwater%20Fish%20Survive%20in%20a%20Warming%20Future&topics=&subtopics=&organizations=

One that doesn't work includes the question mark as %3F:

https://my.usgs.gov/casc-service/search?query=How%20Will%20Coldwater%20Fish%20Survive%20in%20a%20Warming%20Future%3F&topics=&subtopics=&organizations=

If Dell fixes this, all search queries containing those special characters will automatically start working on the live site.
